### PR TITLE
Successful integration of the Statsig SDK

### PR DESCRIPTION
**Pull request by lttsNilesh**
1. StatsigUtil object class for all Statsig SDK APIs.
2. Statsig succesful SDK integration.

### CODE REVIEW

File: android-todo-list/architecture-samples-main/app/build.gradle.kts [hunk 1]
```diff
@@ -25,14 +28,19 @@ plugins {
...
     defaultConfig {
         applicationId = "com.example.android.architecture.blueprints.main"
         minSdk = libs.versions.minSdk.get().toInt()
         targetSdk = libs.versions.targetSdk.get().toInt()
         versionCode = 1
         versionName = "1.0"
 
-        testInstrumentationRunner = "com.example.android.architecture.blueprints.todoapp.CustomTestRunner"
+        testInstrumentationRunner =
+            "com.example.android.architecture.blueprints.todoapp.CustomTestRunner"
```
Comment (daniel-statsig) [inline android-todo-list/architecture-samples-main/app/build.gradle.kts:43]: Can we change the package name to "com.statsig.todoapp"

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import android.app.Application
+import com.statsig.androidsdk.DynamicConfig
+import com.statsig.androidsdk.Statsig
+import com.statsig.androidsdk.StatsigUser
+
+
+/**
+ * Object class StatsIgUtil where all Statsig SDK features will be declared and used.
+ */
+object StatsIgUtil {
```
Comment (daniel-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:12]: This is an invalid formatting of the word Statsig. Please call this class StatsigUtil

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt [hunk 3]
```diff
@@ -31,5 +34,18 @@ class TodoApplication : Application() {
     override fun onCreate() {
         super.onCreate()
         if (BuildConfig.DEBUG) Timber.plant(DebugTree())
+
+        val deviceNameAndId = android.os.Build.MODEL + Settings.Secure.getString(
+            contentResolver,
+            Settings.Secure.ANDROID_ID
+        )
+
+        runBlocking {
```
Comment (daniel-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt:43]: I don't think we should use runBlocking here. Please add a Loading state if needed and only render after the SDK is initialized

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt [hunk 4]
```diff
@@ -31,5 +34,18 @@ class TodoApplication : Application() {
     override fun onCreate() {
         super.onCreate()
         if (BuildConfig.DEBUG) Timber.plant(DebugTree())
+
+        val deviceNameAndId = android.os.Build.MODEL + Settings.Secure.getString(
```
Comment (daniel-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt:38]: What is the purpose of this exactly? Can we not just use a dummy user id like "dummy-user-id"

**Review by daniel-statsig — APPROVED** [review]
Approving to unblock, but I think we should remove the runBlocking in a future commit

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 2 above)
Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:12]: I dont think you need this class at all - its very redundant.  You should just call the statsig methods inline, adding this utils class doesnt do much but increase complexity and verbosity

File: android-todo-list/architecture-samples-main/app/build.gradle.kts (hunk 1 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/build.gradle.kts:43]: Okay I will try to change it to "com.statsig.todoapp"

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 2 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:12]: Okay.

Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:12]: Okay I will remove it.

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt (hunk 4 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt:38]: I wanted to have a unique ID so I can differentiate and use the unique ID for having different rules on them. I will change it to "dummy-user-id".

**Pull request merged**